logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/PersistMessageUseCase.kt

```diff
@@ -45,8 +45,10 @@ internal class PersistMessageUseCaseImpl(
             is MessageContent.RestrictedAsset -> true
             is MessageContent.DeleteForMe -> false
             is MessageContent.Unknown -> false
+            is MessageContent.Availability -> false
+            is MessageContent.FailedDecryption -> true
+            MessageContent.MissedCall -> true
```

Comment:
Whats the difference when using `is` and not using it?

Reply:
Using `is` cast on `when` clauses is only needed for `data class`es objects. However, for plain `object`s, we can simply ignore it. Therefore is up to us to decide whether to add it as a way of keeping visual coherence, or not (as a way to imply that it is a plain `object` class.

Reply:
I just realised that the way I implemented was creating some confusion, so I will opt for adding `is` cast to all types.

Reply:
Oh thanks a lot for the explanation 🙇🏻